packages/layer-composer/src/generators/heatmap/heatmap-animated.ts

```diff
@@ -95,36 +97,53 @@ class HeatmapAnimatedGenerator {
     // TODO - generate this using updated stats API
     const breaks = HARDCODED_BREAKS[config.combinationMode].slice(0, config.tilesets.length)
 
-    const sources = timeChunks.map((timeChunk: TimeChunk) => {
-      const sourceParams: Record<string, string> = {
-        singleFrame: 'false',
-        geomType: config.geomType,
-        filters: config.filters
-          .map((filter, i) => {
-            if (!filter || filter === '') return ''
-            return `filters[${i}]=${filter}`
-          })
-          .join('&'),
-        delta: getDelta(config.start, config.end, timeChunk.interval).toString(),
-        quantizeOffset: timeChunk.quantizeOffset.toString(),
-        interval: timeChunk.interval,
-        numDatasets: config.tilesets.length.toString(),
-        // TODO - generate this using updated stats API
-        breaks: JSON.stringify(breaks),
-        combinationMode: config.combinationMode,
-      }
-      if (timeChunk.start && timeChunk.dataEnd) {
-        sourceParams['date-range'] = [timeChunk.start, timeChunk.dataEnd].join(',')
-      }
-      const url = new URL(`${tilesUrl}?${new URLSearchParams(sourceParams)}`)
-      const source = {
-        id: timeChunk.id,
-        type: 'temporalgrid',
-        tiles: [decodeURI(url.toString())],
-        maxzoom: config.maxZoom,
-      }
-      return source
-    })
+    const sources = flatten(
+      timeChunks.map((timeChunk: TimeChunk) => {
```

Comment:
instead of using two methods and depends on lodash, why not using [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) instead?

Also has a filter method builtin, because returning an empty array will remove the element from the final output

Reply:
Didn't know that, thanks!!